examples/audio_out.c

```diff
@@ -11,7 +11,9 @@
 #include <string.h>
 #include <unistd.h>
 
+#ifdef HAVE_CONFIG_H
 #include <config.h>
+#endif
```

Comment:
Having #ifdef` everywhere is a pain in the neck and almost certain to go wrong sooner of later.

Instead, I would suggest that you replace all existing inlcudes of `config.h` with:
```
#include "src-config.h"
```
and then create that file which would just be:
```
#ifdef HAVE_CONFIG_H
#include "config.h>"
#endif
```

Reply:
What about my suggestion of including it as `"config.h"`? You can then place a file named config.h next to the cpp files including them and be done. No big changes required and guaranteed to work as the current dir is the first to search that include for

Reply:
`config.h` is a generated file, generated by autotools. It can't be generated and part of the repo

> the cpp files

There are no cpp file in this repo.

Reply:
The author wants to include files of this repo in another. There he could add a config.h if he wants to avoid configuring this project.

Reply:
> What about my suggestion of including it as `"config.h"`?

For my specific use case it'll be ok, I can already just put my config.h and set it as a search path.

But `HAVE_CONFIG_H` is common and I think it's better to not make it mandatory.
This is common practice in many libs -

https://github.com/kikinteractive/libpng/blob/b07750220ef71cb4e943560ef0e60430fe3e63a5/pngpriv.h#L57

https://github.com/renesas-rcar/flac_decoder/blob/master/lib/src/libFLAC/crc.c#L45

(and many more)